### PR TITLE
Make joosc compile with latest version of Bazel.

### PR DESCRIPTION
**Pull request by balasanjay**
Recent versions of Bazel have really stepped up their sandboxing game.
Our automatically-included "std.h" no longer works, because Bazel can't
see that include, and errors out when it sees us implicitly depend on
vector/string/etc.

Also, runtime was '#include'-ing some internal Java files, and those
weren't mounted inside the sandbox. Fix this by making it a genrule
instead. Now that these files are normal Java files, use a standard
java_test to test them, and drop the bespoke runtime_tests.sh that
leaves .class files in the tree. 

Also, now that we no longer need to keep this compiling on the school
servers, drop the custom CROSSTOOL in favor of the default one.

### CODE REVIEW

File: lexer/lexer.cpp [hunk 1]
```diff
@@ -219,8 +221,8 @@ bool IsValidIdentifierChar(u8 c) {
 }
 
 bool IsStringEscapable(u8 c) {
-  return c == 'b' | c == 't' | c == 'n' | c == 'f' | c == 'r' | c == '\'' |
-         c == '"' | c == '\\' | IsOctal(c);
+  return c == 'b' || c == 't' || c == 'n' || c == 'f' || c == 'r' || c == '\'' ||
+         c == '"' || c == '\\' || IsOctal(c);
```
Comment (jonathanwei) [inline lexer/lexer.cpp]: wat

File: lexer/lexer.cpp [hunk 2]
```diff
@@ -219,8 +221,8 @@ bool IsValidIdentifierChar(u8 c) {
 }
 
 bool IsStringEscapable(u8 c) {
-  return c == 'b' | c == 't' | c == 'n' | c == 'f' | c == 'r' | c == '\'' |
-         c == '"' | c == '\\' | IsOctal(c);
```
Comment (jonathanwei) [inline lexer/lexer.cpp]: wat

**Comment by jonathanwei** [comment]
LGTM.

**Comment by nelk** [comment]
Nice!

**Pull request merged**